docs/dev/rfc/odfe-cli-rest-api-as-commands.md

```diff
@@ -0,0 +1,298 @@
+# RFC: Support REST api as command
+
+## Objective
+
+This document describes how to support [rest api](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html) in a generic way.
+As of now, elasticsearch support api in different categories like index api, cat api, cluster api, document api, etc to
+configure and access their features. It will take tremendous amount of time and effort to provide every api as native commands.
+Hence, we will be categorizing those REST API based on GET/PUT/POST/DELETE  and allow users to perform their request 
+in a generic way. This will support any future API without any additional support. It also benefits plugin owners while on-boarding their plugin. 
+
+## Available commands
+
+1. [get command](./odfe-cli-rest-api-as-commands.md#1-get-command)
+2. [post command](./odfe-cli-rest-api-as-commands.md#2-post-command)
+3. [put command](./odfe-cli-rest-api-as-commands.md#3-put-command)
+4. [delete command](./odfe-cli-rest-api-as-commands.md#4-delete-command)
+
+## Common parameters
+
+The following options can be applied to all of the odfe-cli commands. These parameters are supported by server as mentioned [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html).
```

Comment:
What do you mean "supported by server"?

Reply:
i was trying to distant our cli with elasticsearch as much as possible. May be it is confusing. Let me revert back to call elasticsearch to avoid this confusion.